doc/Container Hardening/SONiC_container_hardening_HLD.md

```diff
@@ -0,0 +1,374 @@
+		--log-opt max-size=2M --log-opt max-file=5 \
+	{%- endif %}
+
+This will cause the docker file to be altered in the following manner:
+
+**database.sh file**
+
+	docker create --privileged -t -v /etc/sonic:/etc/sonic:ro \ # *Need to remove the --privileged flag*
+		-p 6379:6379 \
+		-e RUNTIME_OWNER=local \
+		--log-opt max-size=2M --log-opt max-file=5 \
+		--tmpfs /tmp \
+		$DB_OPT \
+		$REDIS_MNT \
+		-v /usr/share/sonic/device/$PLATFORM:/usr/share/sonic/platform:ro \
+		--tmpfs /var/tmp \
+		--env "NAMESPACE_ID"="$DEV" \
+		--env "NAMESPACE_PREFIX"="$NAMESPACE_PREFIX" \
+		--env "NAMESPACE_COUNT"=$NUM_ASIC \
+		--name=$DOCKERNAME \
+		docker-database:latest \
+		|| {
+			echo "Failed to docker run" >&1
+			exit 4
+		}
+
+#### Docker privileges
+Removing the root privileges from the docker container - will remove some Linux capabilities that are inherited from the root level permissions.
+
+Running the capabilities list command on a privileged container, this includes all capabilities captured in both [Table 3: Default Linux capabilities](#table-3-default-linux-capabilities) and [Table 4: Extended Linux capabilities](#table-4-extended-linux-capabilities)
+
+	root@ce2c36a0b20c:/# capsh --print
+	Current: = cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read+eip
+
+Running the capabilities list command on an un-privileged container, this includes all capabilities captured in [Table 3: Default Linux capabilities](#table-3-default-linux-capabilities):
+
+	root@ce2c36a0b20c:/# capsh --print
+	Current: cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_mknod,cap_audit_write,cap_setfcap=eip
+
+If, for some reason, a docker must retain a specific capablity functionality on top of the container (which is removed after removing the `--privileged` flag), we can do that with the following:
+
+In the docker-database.mk file adjust this line:
+
+        $(DOCKER_DATABASE)_RUN_OPT += -t –-cap-add NET_ADMIN  # Changed by removing the --privileged flag and adding --cap-add flag
+
+### 4.2 net=host optimization
+
+Here we will provide a detailed example of how to switch from the `--net=host` configuration (host network) to the `--net=bridge` configuration paired with port forwarding in a specific container. We are using the database container as an example for this item.
+
+The original docker creation should be like in the example below:
+
+docker with host sharing:
```

Comment:
>docker with host sharing [](http://example.com/codeflow?start=0&length=24)

As feedback in sonic community review meeting, please considering first unifying single-asic to multi-asic namespace design, and then leverage the current bridge net mode already implemented on multi-asic system.

ref: [https://github.com/sonic-net/sonic-buildimage/pull/3856](https://github.com/sonic-net/sonic-buildimage/pull/3856)

Reply:
Thanks! Will review this and apply the relevant fixes